Riot/ViewController/DirectoryServerPickerViewController.m

```diff
@@ -188,7 +188,7 @@ - (IBAction)onCancel:(id)sender
         onCompleteBlock(nil, nil);
     }
 
-    [self dismissViewControllerAnimated:YES completion:nil];
+    [self.navigationController dismissViewControllerAnimated:YES completion:nil];
```

Comment:
You should use here  `withdrawViewControllerAnimated:completion` defined in the `MXKViewControllerHandling` protocol.